document/value.go

```diff
@@ -250,15 +250,10 @@ func (v Value) MarshalJSON() ([]byte, error) {
 		return strconv.AppendInt(nil, v.V.(int64), 10), nil
 	case DoubleValue:
 		f := v.V.(float64)
-		abs := math.Abs(f)
-		fmt := byte('f')
-		if abs != 0 {
-			if abs < 1e-6 || abs >= 1e21 {
-				fmt = 'e'
-			}
+		if float64(f) == float64(int(f)) {
+			return []byte(strconv.FormatFloat(float64(f), 'f', 1, 64)), nil
 		}
-
-		return strconv.AppendFloat(nil, v.V.(float64), fmt, -1, 64), nil
+		return []byte(strconv.FormatFloat(float64(f), 'f', -1, 64)), nil
```

Comment:
This allocates the formatted value twice, because the string returned by `FormatFloat` is then copied when it's casted to a byte slice. We should use `strconv.AppendFloat` as it's only allocating that memory once.
This is important because this will be called thousands of times per second.

Reply:
Done